lib/services/workflow-api-service.js

```diff
@@ -96,11 +96,8 @@ function workflowApiServiceFactory(
                         definition, configuration.options, context, configuration.domain, true);
             });
         })
-        .tap(function(graph) {
-            var progress = GraphProgress.create(graph, 'Graph "' + graph.name + '" started');
-            return eventsProtocol.publishProgressEvent(graph.instanceId,
-                                                       progress.getProgressEventData());
-        })
+        .tap(taskMessenger.publishGraphStarted)
+        .tap(graphProgressService.publishGraphStarted)
```

Comment:
I see graphFinished related events are generated inside on-taskgraph, it's odd that graphStarted is not within on-taskgraph but instead inside on-http.